### PR TITLE
BE-9367 export error type for use with errors.Is errors.As

### PR DESCRIPTION
**Pull request by waitstime**
For any response from BrainTree that is > 299 status code they return an httpError. So if we export that error that allows the caller to use errors.As and get the status code.

Since I totally mucked up the diff here with merges please see the [original diff here](https://github.com/gametimesf/braintree-go/compare/master...gametimesf:braintree-go:temp?expand=1) All of the rest of the changes are just replacing braintree-go -> gametimesf and then adding CI/CD checks along with fixing lint issues.

### CODE REVIEW

File: response.go [hunk 1]
```diff
@@ -233,13 +233,14 @@ type APIError interface {
...
 
-func (e httpError) StatusCode() int {
+func (e HttpError) StatusCode() int {
 	return int(e)
 }
 
-func (e httpError) Error() string {
+func (e HttpError) Error() string {
 	return fmt.Sprintf("%s (%d)", http.StatusText(e.StatusCode()), e.StatusCode())
 }
```
Comment (waitstime) [inline response.go:245]: here's where they keep the http response code. Exporting this allows us to check what the error code was.

File: response.go [hunk 2]
```diff
@@ -233,13 +233,14 @@ type APIError interface {
 	StatusCode() int
 }
 
-type httpError int
+// HttpError is returned by an API when the status code is 3xx, 4xx, or 5xx and can be used to discern the status code.
+type HttpError int
```
Comment (ajanata) [inline response.go:237]: niiiiiiiit `HTTPError`

**Review by ajanata — COMMENTED** [review]
we only use this in one place, right? might be better to just `replace` in its `go.mod` instead of going through the trouble of changing the import path

**Comment by waitstime** [comment]
> we only use this in one place, right? might be better to just `replace` in its `go.mod` instead of going through the trouble of changing the import path

Yea I think I'll go back to that since the braintree-go is only used directly in cashregister, so that's the only repo the `HTTPError` would need to be checked.

**Review by ajanata — APPROVED** [review]

**Review by Seanskiver — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -1,10 +1,11 @@
 SHELL:=/bin/bash
 
 test:
-	go test -parallel 15 -tags='unit integration' ./...
+	go vet ./...
+	go test --race --cover -tags='unit' ./...
 
 unit:
-	go test -tags=unit ./...
+	go test --race --cover -tags=unit ./...
```
[the remark this answers is not in the document]
Comment (waitstime) [reply Makefile:8]: My goal was consistency in how we typically call `make test` but I changed things back and updated travis to `make unit` instead. The integration tests run against a sandbox env so I didn't want to bother setting that up in CI/CD.

[the remark this answers is not in the document]
Comment (waitstime) [reply Makefile:8]: oops that's fun pushed the update to the wrong branch https://github.com/gametimesf/braintree-go/pull/4

**Pull request merged**